### PR TITLE
Bug 1843610 - Add telemetry for deferred deep link test

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1843610

### CODE REVIEW

**Comment by rocketsroger** [comment]
# Request for data collection review form

All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.

1. What questions will you answer with this data?
  * This data will help us know if user can receive our test deferred deep link and how long does it take to receive it.

2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? 
  * This will allow us to know more about attribution based behaviour.
 
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
  * There are no other alternatives. 
4. Can current instrumentation answer these questions?
  * No. 
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki. 
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
Tab view setting is list or grid | Category 1 - Technical data | https://bugzilla.mozilla.org/show_bug.cgi?id=1843610

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
  * https://dictionary.telemetry.mozilla.org/apps/fenix 
7. How long will this data be collected? Choose one of the following:
  * N/A, collection set to end or be renewed by version 128.

8. What populations will you measure?
  * All channels, all locales, all countries 
9. If this data collection is default on, what is the opt-out mechanism for users?
  * Default Glean SDK opt-out mechanism. 
10. Please provide a general description of how you will analyze this data.
  * Glean and Glam. 
11. Where do you intend to share the results of your analysis?
  * Only on Glean, Glam, and with mobile teams. 
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
   * No third-party tools.

**Review by Amejia481 — APPROVED** [review]

File: fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt [hunk 1]
```diff
@@ -545,6 +547,8 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
                 "finishing" to isFinishing.toString(),
             ),
         )
+
+        PlayStoreAttribution.deferredDeeplinkTime.cancel()
```
Comment (onemahon) [inline fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:551]: Isn't this just measuring the amount of time the Activity is active...? How does this help us understand deferred deep link timing?

Comment (rocketsroger) [reply fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:551]: No cancel means we're not measuring that value.  Where `PlayStoreAttribution.deferredDeeplinkTime.stop()` is where we do the measurement.

This cancel call is just nice so if the user leaves our app we cancel the measurement.

File: fenix/app/src/main/java/org/mozilla/fenix/home/intent/HomeDeepLinkIntentProcessor.kt [hunk 2]
```diff
@@ -61,6 +62,10 @@ class HomeDeepLinkIntentProcessor(
             "settings_privacy" -> GlobalDirections.Settings
             "settings_wallpapers" -> GlobalDirections.WallpaperSettings
             "home_collections" -> GlobalDirections.Home
+            "test_deferred_deep_link" -> {
+                PlayStoreAttribution.deferredDeeplinkTime.stop()
```
Comment (rocketsroger) [inline fenix/app/src/main/java/org/mozilla/fenix/home/intent/HomeDeepLinkIntentProcessor.kt:66]: This is where the measurement happens

File: fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt (hunk 1 above)
Comment (onemahon) [reply fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:551]: (Oh, sorry, just noticed this is a cancel)

Comment (onemahon) [reply fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:551]: haha sorry again, re-commented before refreshing the page

File: fenix/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -146,6 +146,8 @@
                     android:host="urls_bookmarks"/>
                 <data android:scheme="${deepLinkScheme}"
                     android:host="urls_history"/>
+                <data android:scheme="${deepLinkScheme}"
+                    android:host="test_deferred_deep_link"/>
```
Comment (onemahon) [inline fenix/app/src/main/AndroidManifest.xml:150]: I remain a bit skeptical about this, so just trying to validate assumptions: as I understand it, you are hoping that the Play Store will - *after* the user installs the app - initiate an intent with `test_deferred_deep_link` as the `host` part of the URI, thus triggering Fenix to open & capture it with this IntentFilter. Did I summarize that accurately?

Is there any clarity around *how* or *when* the Play Store would initiate such an intent?

Comment (rocketsroger) [reply fenix/app/src/main/AndroidManifest.xml:150]: Yes, that's correct.  Once this change is in a build, we'll have to add `fenix://test_deferred_deep_link` as a deep link in our ongoing Google ad campaign.  That should (hopefully) make Play Store initiate this deep link by trying to start our app with it.  The process should be similar to another app trying to open `fenix://` schemed deep link.

Comment (rocketsroger) [reply fenix/app/src/main/AndroidManifest.xml:150]: One thing to note is I did confirm that if the user is using our app, this particular deep link will not change any of the user experience.  They will not know that they've received a deferred deep link.

**Comment by rocketsroger** [comment]
@Mergifyio backport releases_v116

**Pull request merged**